src/torque-offset-estimator.cpp

```diff
@@ -0,0 +1,265 @@
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.  You should
+ * have received a copy of the GNU Lesser General Public License along
+ * with sot-torque-control.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <sot/torque_control/torque-offset-estimator.hh>
+#include <sot/core/debug.hh>
+#include <dynamic-graph/factory.h>
+#include <sot/torque_control/commands-helper.hh>
+#include <sot/torque_control/motor-model.hh>
+#include <sot/torque_control/hrp2-common.hh>
+#include <Eigen/Dense>
+
+namespace dynamicgraph
+{
+  namespace sot
+  {
+    namespace torque_control
+    {
+
+#define ALL_INPUT_SIGNALS m_base6d_encodersSIN << m_accelerometerSIN << m_gyroscopeSIN \
+                          << m_jointTorquesSIN
+
+#define ALL_OUTPUT_SIGNALS m_jointTorquesEstimatedSOUT
+
+      namespace dynamicgraph = ::dynamicgraph;
+      using namespace dynamicgraph;
+      using namespace dynamicgraph::command;
+      using namespace Eigen;
+
+      /// Define EntityClassName here rather than in the header file
+      /// so that it can be used by the macros DEFINE_SIGNAL_**_FUNCTION.
+      typedef TorqueOffsetEstimator EntityClassName;
+      typedef int dummy;
+      std::vector<Eigen::VectorXd,Eigen::aligned_allocator<Eigen::VectorXd> > stdAlignedVector;
+      /* --- DG FACTORY ------------------------------------------------------- */
+      DYNAMICGRAPH_FACTORY_ENTITY_PLUGIN(TorqueOffsetEstimator,"TorqueOffsetEstimator");
+
+      /* --- CONSTRUCTION ----------------------------------------------------- */
+      /* --- CONSTRUCTION ----------------------------------------------------- */
+      /* --- CONSTRUCTION ----------------------------------------------------- */
+      TorqueOffsetEstimator::
+      TorqueOffsetEstimator( const std::string & name )
+        : Entity(name),
+          CONSTRUCT_SIGNAL_IN(base6d_encoders,      dynamicgraph::Vector)
+        ,CONSTRUCT_SIGNAL_IN(accelerometer,         dynamicgraph::Vector)
+        ,CONSTRUCT_SIGNAL_IN(gyroscope,             dynamicgraph::Vector)
+        ,CONSTRUCT_SIGNAL_IN(jointTorques,          dynamicgraph::Vector)
+        ,CONSTRUCT_SIGNAL_INNER(collectSensorData,  dynamicgraph::Vector, ALL_INPUT_SIGNALS)
+        ,CONSTRUCT_SIGNAL_OUT(jointTorquesEstimated,
+                              dynamicgraph::Vector, m_collectSensorDataSINNER)
+        ,sensor_offset_status(PRECOMPUTATION)
+      {
+        Entity::signalRegistration( ALL_INPUT_SIGNALS << ALL_OUTPUT_SIGNALS);
+        addCommand("init", makeCommandVoid2(*this, &TorqueOffsetEstimator::init,
+                                            docCommandVoid2("Initialize the estimator" ,
+                                                            "urdfFilePath",
+                                                            "Definition of the imu in the chest frame")));
```

Comment:
This is not clear, I suggest replacing it with something like "Homogeneous transformation from chest frame to IMU frame"